pkg/util/ansible_execution.go

```diff
@@ -59,20 +59,21 @@ func AnsibleExecution(
 
 	ansibleEEMounts := storage.VolMounts{}
 
-	ansibleEE, err := GetAnsibleExecution(ctx, helper, obj, service.Name)
+	ansibleEE, err := GetAnsibleExecution(ctx, helper, obj, service.Name, targetNodeset)
 	if err != nil && !k8serrors.IsNotFound(err) {
 		return err
 	}
 	if ansibleEE == nil {
 
-		executionName := fmt.Sprintf("%s-%s", GetAnsibleExecutionNamePrefix(service), obj.GetName())
+		executionName := fmt.Sprintf("%s-%s-", GetAnsibleExecutionNamePrefix(service), targetNodeset)
```

Comment:
why the extra `-`?

Reply:
Otherwise the generated name will end up with the random part concatenated directly to the nodeset name. 
Which can get confusing, as the random suffix can include `s`. For example `mysuperservice-defaultnode-syba` would turn into `mysuperservice-defaultnodesyba`.

Is it plural `nodes` or singular `node`? Who can say. This way it's more obvious which part is generated and which is predetermined.

Reply:
`obj` is the Deployment.

Don't we actually need the AnsibleEE name to be a combination of Service, NodeSet, and Deployment? Otherwise, just using Service and NodeSet might find completed AnsibleEE's from a previous Deployment wouldn't it?

Reply:
I'll rename the var to make it clear. The function is being used for a single purpose anyway, so it's not like we are going to be supplying other objects to it.

Reply:
New name is now set by all three.